numba/tests/test_numbers.py

```diff
@@ -0,0 +1,90 @@
+# (Some) Tests for targets/numbers.py
+from __future__ import print_function, absolute_import, division
+
+import numpy as np
+
+from numba import njit, types
+from numba.errors import TypingError
+from .support import TestCase
+
+
+def gen_view(a,b):
+    def impl(x):
+        return a(x).view(b)
+    return impl
+
+
+class TestViewIntFloat(TestCase):
+    """ This tests the 'view' method on NumPy scalars. """
+
+    def do_testing(self, inputs, dtypes):
+        for value, initial_type, expected in inputs:
+            for target_type, result in zip(dtypes, expected):
+                view = njit(gen_view(initial_type, target_type))
+                if not np.isnan(result):
+                    # check against predefined value
+                    self.assertEqual(view(value), target_type(result))
+                    # check against numpy
+                    self.assertEqual(view(value),
+                                     initial_type(value).view(target_type))
```

Comment:
You can just do `view.py_func(value)` the jit dispatcher keeps reference to the original function as the `.py_func` member, this *explicitly* tests interpretation of the same code Numba is compiling.

Reply:
Ohhhh, that is handy! Good to know.